process/block/baseProcess.go

```diff
@@ -1239,3 +1242,66 @@ func (bp *baseProcessor) addHeaderIntoTrackerPool(nonce uint64, shardID uint32)
 		bp.blockTracker.AddTrackedHeader(headers[i], hashes[i])
 	}
 }
+
+func (bp *baseProcessor) commitTrieEpochRootHashIfNeeded(metaBlock *block.MetaBlock) error {
+	trieEpochRootHashStorageUnit := bp.store.GetStorer(dataRetriever.TrieEpochRootHashUnit)
+	if check.IfNil(trieEpochRootHashStorageUnit) {
+		return nil
+	}
+	_, isStorerDisabled := trieEpochRootHashStorageUnit.(*storageUnit.NilStorer)
```

Comment:
I think we can get rid of this check. If the storer is nil, it will return on line 1249.

Reply:
no.. line 1249 returns a nil/disabled storer so this check is mandatory. An alternative would have been adding a function `IsNilStorer() bool` inside the `Storer` interface but the changeset would have been too big